### DISCUSSION TITLE
What's difference between Instrument.AspNetCoreRequests() and Instrument.HttpClientRequests()

### DISCUSSION THREAD

**Discussion by nabeel-servcorp** [datalust/seq-tickets#2150] [Help]
What is the difference between `Instrument.HttpClientRequests()` and ` Instrument.AspNetCoreRequests()` methods? If we use ` Instrument.AspNetCoreRequests()` in our application, is there even a need to use `Instrument.HttpClientRequests()`? What about the vice versa?

**Comment by nblumhardt** [comment]
Hi! The `AspNetCoreRequests` one is server-side and records requests received by an ASP.NET Core app. The `HttpClientRequests` one (enabled by default) is client-side and records requests made through the `HttpClient` type. Hope this helps!

**Reply by nabeel-servcorp** [reply]
What does "_enabled by default_" mean? Does "_enabled by default_" mean that `.Instrument.HttpClientRequests()` statement is unnecessary and redundant in the below snippet?

```c#
using var _ = new ActivityListenerConfiguration()
    .Instrument.AspNetCoreRequests()
    .Instrument.SqlClientCommands()
    .Instrument.HttpClientRequests()
    .TraceToSharedLogger();
```

**Reply by nblumhardt** [reply]
Yes, that's correct - it's not necessary to add that line.
